web/ui/react-app/src/Graph.tsx

```diff
@@ -160,45 +157,46 @@ class Graph extends PureComponent<GraphProps> {
 
       if (i % colorPoolSize === 0 && i) {
         if (variation >= 0) {
-          if (variation < 0.5) {
-            variation = -variation - 0.2;
-          } else variation = 0;
-        } else variation = -variation;
+          variation = variation < 0.5 ? -variation - 0.2 : 0;
+        } else {
+          variation = -variation;
+        }
       }
-
-      colors[i] = c.scale('rgb', 1 + variation);
-    }
-
-    return colors;
+      return $.color.parse(colorPool[i % colorPoolSize] || '#666').scale('rgb', 1 + variation);
+    });
   }
 
-  getData() {
+  getData(): GraphSeries[] {
     const colors = this.getColors();
-
-    return this.props.data.result.map((ts: any /* TODO: Type this*/, index: number) => {
+    const { hoveredSeriesIndex } = this.state;
+    const { stacked, queryParams } = this.props;
+    const { startTime, endTime, resolution } = queryParams!;
+    return this.props.data.result.map((ts, index) => {
       // Insert nulls for all missing steps.
       const data = [];
       let pos = 0;
-      const params = this.props.queryParams!;
 
-      for (let t = params.startTime; t <= params.endTime; t += params.resolution) {
+      for (let t = startTime; t <= endTime; t += resolution) {
         // Allow for floating point inaccuracy.
-        if (ts.values.length > pos && ts.values[pos][0] < t + params.resolution / 100) {
-          data.push([ts.values[pos][0] * 1000, this.parseValue(ts.values[pos][1])]);
+        const currentValue = ts.values[pos];
+        if (ts.values.length > pos && currentValue[0] < t + resolution / 100) {
+          data.push([currentValue[0] * 1000, this.parseValue(currentValue[1])]);
           pos++;
         } else {
           // TODO: Flot has problems displaying intermittent "null" values when stacked,
           // resort to 0 now. In Grafana this works for some reason, figure out how they
           // do it.
-          data.push([t * 1000, this.props.stacked ? 0 : null]);
+          data.push([t * 1000, stacked ? 0 : null]);
         }
       }
+      const { r, g, b } = colors[index];
 
       return {
         labels: ts.metric !== null ? ts.metric : {},
-        data: data,
-        color: colors[index],
-        index: index,
+        color: `rgba(${r}, ${g}, ${b}, ${hoveredSeriesIndex === null || hoveredSeriesIndex === index ? 1 : 0.5})`,
```

Comment:
Playing with it a bit more, let's do `0.5` -> `0.3` to make highlighted series stand out even more?